src/docker-in-docker/install.sh

```diff
@@ -468,6 +469,29 @@ if [ "${INSTALL_DOCKER_BUILDX}" = "true" ]; then
     find "${docker_home}" -type d -print0 | xargs -n 1 -0 chmod g+s
 fi
 
+DOCKER_DEFAULT_IP6_TABLES=""
+requested_version=""
+# checking whether the version requested either is in semver format or just a number denoting the major version
+# extracting the major version number out of the two scenarios
+semver_regex="^(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(?:-([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?(?:\+([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?$"
```

Comment:
Why do we need such a complex semver? Why can't we use `^(latest|none|lts|[0-9]+)`?

Reply:
ok will change..

Reply:
this semver regex checks for entire semver i.e. 1.2.3 pattern, which will confirm if semver has actually been passed so can pick the major version from it